currency/translation.go

```diff
@@ -20,3 +20,83 @@ var translations = map[*Item]Code{
 	XDG.Item:  DOGE,
 	USDT.Item: USD,
 }
+
+// NewTranslations returns a new translation map, the key indicates the exchange
+// representation and the value indicates the internal representation/common/standard
+// representation. e.g. XBT as key and BTC as value, this is useful for exchanges
+// that use different naming conventions.
+func NewTranslations(t map[Code]Code) Translations {
+	lookup := make(map[*Item]Code)
+	for k, v := range t {
+		lookup[k.Item] = v
+	}
+	return lookup
+}
+
+// Translations is a map of translations for a specific exchange implementation
+type Translations map[*Item]Code
+
+// Translate returns the translated currency code, usually used to convert
+// exchange specific currency codes to common currency codes. If no translation
+// is found it will return the original currency code.
+func (t Translations) Translate(incoming Code) Code {
+	if len(t) == 0 {
+		return incoming
+	}
+	val, ok := (t)[incoming.Item]
+	if !ok {
+		return incoming
+	}
+	return val
+}
+
+// Translator is an interface for translating currency codes
+type Translator interface {
+	// TODO: Add a asset.Item param so that we can translate for asset
+	// permutations. Also return error.
+	Translate(Code) Code
+}
+
+// PairsWithTranslation is a pair list with a translator for a specific exchange.
+type PairsWithTranslation struct {
+	Pairs      Pairs
+	Translator Translator
+}
+
+// keyPair defines an immutable pair for lookup purposes
+type keyPair struct {
+	Base  *Item
+	Quote *Item
+}
```

Comment:
Add this to the key package as `key.Pair`

Reply:
I think that might be a cycle import issue but I will give it a crack.

Reply:
Oh yeah, that's okay then